x/btcstkconsumer/keeper/msg_server_test.go

```diff
@@ -22,3 +29,82 @@ func TestMsgServer(t *testing.T) {
 	require.NotNil(t, ctx)
 	require.NotEmpty(t, k)
 }
+
+func FuzzRegisterConsumer(f *testing.F) {
+	datagen.AddRandomSeedsToFuzzer(f, 10)
+
+	f.Fuzz(func(t *testing.T, seed int64) {
+		r := rand.New(rand.NewSource(seed))
+
+		babylonApp, ctx := wasmtest.SetupAppWithContext(t)
+		bscKeeper := babylonApp.BTCStkConsumerKeeper
+		msgServer := keeper.NewMsgServerImpl(bscKeeper)
+
+		/*
+			Test gov-gated registration
+		*/
+		// enable gov-gated registration
+		err := bscKeeper.SetParams(ctx, types.Params{
+			PermissionedIntegration: true,
+		})
+		require.NoError(t, err)
+		// generate a random consumer register
+		consumerRegister := datagen.GenRandomCosmosConsumerRegister(r)
+		// Register the consumer
+		_, err = msgServer.RegisterConsumer(ctx, &types.MsgRegisterConsumer{
+			ConsumerId:          consumerRegister.ConsumerId,
+			ConsumerName:        consumerRegister.ConsumerName,
+			ConsumerDescription: consumerRegister.ConsumerDescription,
+		})
+		require.Error(t, err)
+		require.ErrorIs(t, err, govtypes.ErrInvalidSigner)
+		// disable gov-gated registration
+		err = bscKeeper.SetParams(ctx, types.Params{
+			PermissionedIntegration: false,
+		})
+		require.NoError(t, err)
+
+		/*
+			Test registering Cosmos consumer
+		*/
+		// generate a random consumer register
+		consumerRegister = datagen.GenRandomCosmosConsumerRegister(r)
+		// mock IBC light client
+		babylonApp.IBCKeeper.ClientKeeper.SetClientState(ctx, consumerRegister.ConsumerId, &ibctmtypes.ClientState{})
+		// Register the consumer
+		_, err = msgServer.RegisterConsumer(ctx, &types.MsgRegisterConsumer{
+			ConsumerId:          consumerRegister.ConsumerId,
+			ConsumerName:        consumerRegister.ConsumerName,
+			ConsumerDescription: consumerRegister.ConsumerDescription,
+		})
+		require.NoError(t, err)
+		// check that the consumer is registered
+		consumerRegister2, err := bscKeeper.GetConsumerRegister(ctx, consumerRegister.ConsumerId)
+		require.NoError(t, err)
+		require.Equal(t, consumerRegister.String(), consumerRegister2.String())
+
+		/*
+			Test registering ETH L2 consumer
+		*/
+		// mock the wasm contract
+		contractAddr := mockSmartContract(t, ctx, babylonApp)
+		// generate a random consumer register
+		consumerRegister = datagen.GenRandomETHL2Register(r, contractAddr.String())
+		// Register the consumer
+		_, err = msgServer.RegisterConsumer(ctx, &types.MsgRegisterConsumer{
+			ConsumerId:                   consumerRegister.ConsumerId,
+			ConsumerName:                 consumerRegister.ConsumerName,
+			ConsumerDescription:          consumerRegister.ConsumerDescription,
+			EthL2FinalityContractAddress: contractAddr.String(),
```

Comment:
At some point it would be good to check that the contract is of a particular type / adheres to some particular API.

Reply:
Good idea, added